### DISCUSSION TITLE
Can an extension be configured?

### DISCUSSION THREAD

**Discussion by jayped007** [mlocati/docker-php-extension-installer#607] [Q&A] [answered]
Is it possible to configure an extension as part of the install of that extension, for example I have a case where the following configuration was used prior to install of the extension:

     docker-php-ext-configure gd \
         --enable-gd \
         --with-freetype \
         --with-webp \
         --with-jpeg &&
     docker-php-ext-install gd && \
     docker-php-ext-enable gd

If not, is it then necessary to separately perform this for specific extensions that you want to configure?  Is there any example of doing this?

**Comment by mlocati** [comment, marked as answer]
It's not possible to configure the extensions (except for [a few cases](https://github.com/mlocati/docker-php-extension-installer/blob/master/README.md#configuration)).

BTW please remark that, generally speaking, I tried to enable all the features of the extensions.

For example, for GD we use the following configuration options:

- for PHP 5.5/5.6:
  `--with-gd --with-jpeg-dir --with-png-dir --with-zlib-dir --with-xpm-dir --with-freetype-dir --enable-gd-native-ttf --with-vpx-dir`
- for PHP 7.0/7.1:
  `--with-gd --with-jpeg-dir --with-png-dir --with-zlib-dir --with-xpm-dir --with-freetype-dir --enable-gd-native-ttf --with-webp-dir`
- for PHP 7.2/7.3:
  `--with-gd --with-jpeg-dir --with-png-dir --with-zlib-dir --with-xpm-dir --with-freetype-dir --with-webp-dir`
- for PHP 7.4/8.0:
  `--enable-gd --with-webp --with-jpeg --with-xpm --with-freetype`
- for PHP 8.0/8.1:
  - if `IPE_GD_WITHOUTAVIF` is set:
    `--enable-gd --with-webp --with-jpeg --with-xpm --with-freetype --with-avif`
  - if `IPE_GD_WITHOUTAVIF` is not set:
    `--enable-gd --with-webp --with-jpeg --with-xpm --with-freetype`

**Reply by jayped007** [reply]
Thank you!  Explained both awesomely and quickly!

**Reply by pawel-srebniak-sirocco** [reply]
Thanks, saved me hours of works with explaining this for php 8.1 -> it also works with 8.3
